holmes/core/investigation_structured_output.py

```diff
@@ -58,7 +64,101 @@ def combine_sections(sections: Dict) -> str:
     return content
 
 
-def process_response_into_sections(response: Any) -> Tuple[str, OutputSectionsDataType]:
+def parse_markdown_into_sections_from_equal_sign(
+    markdown_content: str,
+) -> Optional[Dict[str, Optional[str]]]:
+    """Splits a markdown in different sections where the key is a top level title underlined with `====` and the value is the content
+    ```
+    Header Title
+    ===========
+    Content here
+    ```
+    =>
+    {
+      "Header Title": "Content here"
+    }
+    """
+    matches = re.split(r"(?:^|\n)([^\n]+)\n=+\n", markdown_content.strip())
+
+    # Remove any empty first element if the text starts with a header
+    if matches[0] == "":
+        matches = matches[1:]
+
+    sections = {}
+
+    for i in range(0, len(matches), 2):
+        if i + 1 < len(matches):
+            header = matches[i]
+            content = matches[i + 1].strip()
+            sections[header] = content
+
+    if len(sections) > 0:
+        return sections
+    else:
+        return None
+
+
+def parse_markdown_into_sections_from_hash_sign(
+    markdown_content: str,
+) -> Optional[Dict[str, Optional[str]]]:
+    """Splits a markdown in different sections where the key is a top level title underlined with `====` and the value is the content
+    ```
+    # Header Title
+    Content here
+    ```
+    =>
+    {
+      "Header Title": "Content here"
+    }
+    """
+    # Split the text into sections based on headers (# Section)
+    matches = re.split(r"\n(?=# )", markdown_content.strip())
```

Comment:
Just raising edge case here, what if the response won't start with new line? this could happen?